go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -31,12 +31,20 @@ func buildGeneralDDLPlan(sql string, ddlStatement sqlparser.DDLStatement, vschem
 		return nil, err
 	}
 
+	if ddlStatement.IsTemporary() {
+		if normalDDLPlan.Keyspace.Sharded {
+			return nil, vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "Temporary table not supported in sharded keyspace: %s", normalDDLPlan.Keyspace.Name)
+		}
+		onlineDDLPlan = nil // emptying this so it does not accidentally gets used somewhere
+	}
+
 	return &engine.DDL{
-		Keyspace:  normalDDLPlan.Keyspace,
-		SQL:       normalDDLPlan.Query,
-		DDL:       ddlStatement,
-		NormalDDL: normalDDLPlan,
-		OnlineDDL: onlineDDLPlan,
+		Keyspace:        normalDDLPlan.Keyspace,
+		SQL:             normalDDLPlan.Query,
+		DDL:             ddlStatement,
+		NormalDDL:       normalDDLPlan,
+		OnlineDDL:       onlineDDLPlan,
+		CreateTempTable: ddlStatement.IsTemporary(),
```

Comment:
It feels like a bit of high level location to place this low level information, but I see why this is necessary.